### PR TITLE
Fix browser behaviour when homescreen tab is tapped - do not open new…

### PR DESCRIPTION
**Pull request by Vishwa-Mozilla**
… tab when tab is already present in the browser

---

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/intent/src/main/java/mozilla/components/feature/intent/processing/TabIntentProcessor.kt [hunk 1]
```diff
@@ -50,8 +50,9 @@ class TabIntentProcessor(
         return if (url.isNullOrEmpty()) {
             false
         } else {
-            val session = createSession(url, private = isPrivate, source = Source.ACTION_VIEW)
-            loadUrlUseCase(url, session, LoadUrlFlags.external())
+            sessionManager.sessions.find { it.url == url }?.also { sessionManager.select(it) }
```
Comment (NotWoods) [inline components/feature/intent/src/main/java/mozilla/components/feature/intent/processing/TabIntentProcessor.kt:53]: I think this is a little hard to read as a one liner. In this case it's better to break up the logic a little, something like:
```kt
val existingSession = sessionManager.sessions.find { it.url == url }
if (existingSession != null) {
  ...
} else {
  ...
}
```

**Comment by jonalmeida** [comment]
bors r-

Could we please squash these commits into one before merging?  🙂 

You can also follow the format: `Close #[issue number]: A short line of the bug fix`.

**Comment by ekager** [comment]
bors r=NotWoods

**Comment by ekager** [comment]
bors cancel

**Comment by NotWoods** [comment]
For future reference, you can link to issues in other repositories like this: mozilla-mobile/fenix#5201 (but its better to file an issue here too)

**Comment by ekager** [comment]
We just opened #7067 and will link that one now!

**Comment by jonalmeida** [comment]
> For future reference, you can link to issues in other repositories like this: [mozilla-mobile/fenix#5201](https://github.com/mozilla-mobile/fenix/issues/5201) (but its better to file an issue here too)

We avoid linking to external app codebases and issues explicitly since we build for multiple consumers.

**Comment by ekager** [comment]
I think we are good now! Linked to an AC issue in the commit

bors r=NotWoods

**Pull request merged**